src/persist-client/src/critical.rs

```diff
@@ -331,12 +331,11 @@ where
         }
     }
 
-    /// Politely expires this reader, releasing its since capability.
-    #[instrument(level = "debug", fields(shard = %self.machine.shard_id()))]
-    pub async fn expire(mut self) {
-        let (_, maintenance) = self.machine.expire_critical_reader(&self.reader_id).await;
-        maintenance.start_performing(&self.machine, &self.gc);
-    }
+    // Expiry temporarily removed.
+    // If you'd like to stop this handle from holding back the since of the shard,
+    // downgrade it to [].
+    // TODO(bkirwi): revert this when since behaviour on expiry has settled,
+    // or all readers are associated with a critical handle.
```

Comment:
Why did we have to remove this?

Reply:
Before this change, expiring a handle and downgrading the since to [] were more or less equivalent. After this change, though, they would have quite different behaviour. (Because, if we don't have any critical handles registered, we'll  default to the current since.)

Conveniently, nobody expires critical handles at the moment, so this doesn't matter. Removing the method means we don't have to worry about this case cropping up during the migration.

Reply:
Just double checking, we're removing this because it would allow the following to occur:

1. Shard with a `since: 8`
2. Critical Reader "A" created with a `since: 10`
3. Leased Reader "B" created, inherits `since: 10`
4. Critical Reader "A" expired
5. Leased Reader "C" created, inherits `since: 8` from the shard

This isn't _incorrect_ necessarily because the shard is readable since `8`, but pretty weird behavior so we'll punt on it for now?

Reply:
Yeah, that's right. Experiments suggest that changing how since holds work is going to be a _very_ delicate migration, so the fewer interactions we need to worry about...